### ISSUE TITLE
Delayed plugin initialisation

### ISSUE THREAD

**Issue by eveningkid** [manyuanrong/deno_mongo#55]
Hey,

First of all, thanks for all the efforts on this, I like deno_mongo very much :)

Now, here's about the issue I'm encountering. I am trying to lazy-load `deno_mongo` for my project `denodb` which handles different types of databases. The problem being that if someone only wants to use mysql, the library shouldn't load mongodb (which comes with additional command flags `unstable` and `allow-plugin` at least).

After digging in the code, I found a solution to lazy-load everything:
```typescript
const { MongoClient, RELEASE_URL, init } = await import("../../unstable_deps.ts");
await init(RELEASE_URL);
const client = new MongoClient();
```

But obviously this is when I realised, this is already being done in `mod.ts`. The problem is, when lazy-loaded, the module doesn't wait on the `await init(...)` instruction from that file.

```bash
Uncaught Error: The plugin must be initialized before use
```

So far, the only solution I found was to remove that line (with `init`) from my local package `mod.ts`. Otherwise, by keeping my additional `await init(...)`, I am getting an error saying that I am trying to initialise the same plugin twice (which makes sense).

I am therefore asking: could something be done here so that this library could be lazy-loaded when necessary? I know this might only be my problem right now, but I am sure it might be someone else's someday.

Just let me know if you could do something about this, any thought.

Thanks for your time and keep up the good work :)

**Comment by manyuanrong** [comment]
At the beginning, init was not included in mod.ts, until many people forgot to call init before using it, it exists in mod.ts.

One of the solutions is to use dynamic import to load the deno_mongo library in your integrated library

**Comment by eveningkid** [comment]
Thanks for the quick reply :)

Uh oh, what a pity! I am actually already lazy-loading it:
```typescript
// unstable_deps.ts
export {
  MongoClient as MongoDBClient,
  ClientOptions as MongoDBClientOptions,
  Database as MongoDBDatabase,
  init as initMongoDBPlugin,
  RELEASE_URL as MONGODB_PLUGIN_RELEASE_URL,
} from "https://deno.land/x/mongo@v0.8.0/mod.ts";

// other-file.ts, lazy-loaded
const { MongoDBClient, MONGODB_PLUGIN_RELEASE_URL, initMongoDBPlugin } =
  await import("../../unstable_deps.ts");
```

Unless I am doing something wrong here?

So basically when I do this, then create a client, it raises the following error:

```bash
Uncaught Error: The plugin must be initialized before use
```

Just as if the `await init(...)` command in your `mod.ts` was not done yet. When I manually call `init(...)` it says both commands are running which causes another issue.

My feeling is that when lazy-loaded, the compiler doesn't wait on the `await` statement at the bottom file.

I still believe that this shouldn't run by default, even if people forget about it...

**Comment by divy-work** [comment]
Lady loading of deno plugins is a bad idea in terms of the download time it might take.

**Comment by eveningkid** [comment]
Hey Divy,

Lazy loading isn't a bad idea.

My project is using multiple librairies and `deno_mongo` is the only unstable dependency. This means two things:
- if you **don't** use the part of my project using mongo, you will still need the `--unstable` flag and others, every time.
- you load things a user **might not ever need**.

Downloading time happens in the initialisation, once, then is cached. We are talking seconds here.

Lazy loading could be a bad idea in some cases, not in mine.

All we are talking here however is how something that could have been lazy-loaded in some useful cases cannot anymore, only because some users wouldn't use the library properly.

I have seen very few librairies forcing the initialisation as it usually happens through a manual function call so there is no "hidden magic" and so it makes the developer in charge of what happens and when.

Coming out of the blue saying it is a bad idea because of download time, is not helping in any way.

**Comment by manyuanrong** [comment]
The project was rewritten using TS. This will no longer be a issue
